### ISSUE TITLE
Improve Release Notes generation

### ISSUE THREAD

**Issue by ericvn** [istio/istio#23622]
What can be done to help the release managers better scrape and create the Release Notes.

**Comment by brian-avery** [comment]
There's prior work here from the Kubernetes release notes group that looks like it might be really good for us to follow:https://github.com/kubernetes-sigs/release-notes

They have a tool that scans the commits between two releases. This generates release notes and upgrade notes content based on flags in pull requests. This content is then collected in a pull request which is reviewed by their release notes group and docs group.

To create an upgrade note, add something similar to the following to a pull request:
![image](https://user-images.githubusercontent.com/8484260/83155704-82b81d00-a0cf-11ea-8592-257d9b39478a.png)

To create a release notes, add something similar to the following to a pull request: 

![image](https://user-images.githubusercontent.com/8484260/83155775-95caed00-a0cf-11ea-9599-d9e33d6a3215.png)

An example release notes PR is: 
https://github.com/kubernetes/sig-release/pull/1091/files

I think it would be useful to record: 
* release/upgrade notes comment
* issue ID (we've always had trouble keeping track of these, but if we generate them automatically, this step should be easy)

I would also call out what's going in release notes/upgrade notes a bit more. Perhaps: 

Release notes: 
``````
*Fixed* foo to do bar
``````

Upgrade notes: 
``````
We've removed the *foo* utility. Please upgrade to the *bar* utility. 
``````

I haven't taken a look at this yet, but the release notes team also has a talk that they did on their release notes process: https://www.youtube.com/watch?v=qxRVnEFCYlo&feature=youtu.be&t=19142

**Comment by brian-avery** [comment]
An example of what the pull request template might look like is here: https://github.com/brian-avery/istio/pull/2

**Comment by brian-avery** [comment]
The current status of this is that we now have: 
* A YAML specification for defining release notes on a per pull request basis. 
* A prow job that verifies that either release notes are present or a maintainer has explicitly said that the pull request has no user facing changes. 
* Markdown templates that specify the overall structure for release notes along with a rule of what release notes should go where. 
* A utility in the tools repo to generate release notes. 

While this tooling isn't yet complete, the original goal of this was to "improve release notes generation" and we have achieved that. We have shortened the feedback loop for release notes and made it easier to generate the actual release notes. Further work will continue in 1.8 under https://github.com/istio/istio/issues/26365
